tests/PHPStan/Analyser/nsrt/narrow-cast.php

```diff
@@ -63,6 +63,12 @@ function castInt($x, string $s, bool $b) {
         assertType('false', $b);
     }
 
+	if ((int) $s) {
+		assertType('string', $s); // could be *NEVER*
```

Comment:
reachable for `1` or `01`, `numeric-string~non-falsy-string`?

Reply:
thank you

Reply:
we reach ELSE for `'0.1'` which is a numeric string

Reply:
and not for `1.5` - https://3v4l.org/VBPn2 - so `string` is best

also notice `1x` - https://3v4l.org/PGs9f

Reply:
`string~(''|'0')` is `non-falsy-string`. but I think we are in crazy territory land where `string` should be good enough ;)